### PR TITLE
[Feature] add locales CP to the service

### PR DESCRIPTION
**Pull request by kellyselden**
A result of #247. Adds a way to get a list of the added locales. Useful for rendering this list to a template to switch locales.

### CODE REVIEW

File: addon/get-locales.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+const matchKey = '/locales/(.+)/translations$';
+
+export default function getLocales() {
+  return Ember.keys(requirejs.entries).reduce((locales, module) => {
+    var match = module.match(matchKey);
+    if (match) {
+      locales.push(match[1]);
+    }
+    return locales;
```
Comment (jamesarosen) [inline addon/get-locales.js]: This should probably wrap the result in `Ember.A()`. I think that's the best practice for things that might end up in a Glimmer `each`.

File: tests/unit/locales-test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  assert.equal(i18n.get('locales').length, locales.length);
+});
+
+test('locales have the correct names', function(assert) {
+  const i18n = this.subject();
+
+  assert.expect(locales.length);
+
+  locales.forEach(locale => {
```
Comment (jamesarosen) [inline tests/unit/locales-test.js]: You could use `assert.deepEqual` here, though you might have to `sort` the results.

File: addon/get-locales.js (hunk 1 above)
Comment (kellyselden) [inline addon/get-locales.js]: I went back and forth on that. I will add it.

File: tests/unit/locales-test.js (hunk 2 above)
Comment (kellyselden) [inline tests/unit/locales-test.js]: Good call. Had to reorder the expected.

**Comment by kellyselden** [comment]
fixed the comments.

File: addon/get-locales.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+const matchKey = '/locales/(.+)/translations$';
+
+export default function getLocales() {
+  return Ember.keys(requirejs.entries).reduce((locales, module) => {
+    var match = module.match(matchKey);
+    if (match) {
+      locales.pushObject(match[1]);
+    }
+    return locales;
+  }, Ember.A());
```
Comment (jamesarosen) [inline addon/get-locales.js]: Nice use of `reduce`!

**Comment by jamesarosen** [comment]
:octopus: :moneybag: :tangerine:

File: addon/get-locales.js (hunk 3 above)
Comment (kellyselden) [inline addon/get-locales.js]: Thanks!

**Pull request merged**